build-logic/gobley-gradle-uniffi/src/main/java/gobley/gradle/uniffi/utils/TakeIfTransformer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package gobley.gradle.uniffi.utils;
+
+import org.gradle.api.Transformer;
+
+import java.util.function.Predicate;
+
+import javax.annotation.Nonnull;
+
+import gobley.gradle.InternalGobleyGradleApi;
+
+// The Kotlin equivalent of the following code compiles well, but Android Studio shows an error.
```

Comment:
What's the Android Studio error?

Reply:
`Transformer` is intended to allow returning `null`, but doing so makes Android Studio emit a "you're putting `null` to a non-null context" error. I'll revert this change and open a new issue.